hydra-node/src/Hydra/API/Server.hs

```diff
@@ -26,15 +27,30 @@ data APIServerLog
   | APIInvalidRequest {receivedRequest :: Text}
   deriving (Eq, Show)
 
+withAPIServer ::
+  Tx tx =>
+  IP ->
+  PortNumber ->
+  Tracer IO APIServerLog ->
+  (ClientRequest tx -> IO ()) ->
+  ((ClientResponse tx -> IO ()) -> IO ()) ->
+  IO ()
+withAPIServer host port tracer requests node = do
+  responseChannel <- newBroadcastTChanIO
+  let sendResponse = atomically . writeTChan responseChannel
+  race_
+    (runAPIServer host port tracer requests responseChannel)
+    (node sendResponse)
```

Comment:
This `node` confused me at first for it's a bit "out of context" / assuming the function is used in some particular context, but okay.